### PR TITLE
Add MRO registration to Ruby to support Rule Types

### PR DESCRIPTION
**Pull request by leina05**
PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

File: languages/ruby/lib/oso/polar/host.rb [hunk 1]
```diff
@@ -50,7 +50,7 @@ def initialize(name:, klass:, id:, fields:, fetcher:)
 
     # Translate between Polar and the host language (Ruby).
     class Host # rubocop:disable Metrics/ClassLength
-      # @return [Hash<String, Class>]
+      # @return [Hash<String, UserType>]
```
Comment (gj) [inline languages/ruby/lib/oso/polar/host.rb:53]: wot

Comment (dhatch) [reply languages/ruby/lib/oso/polar/host.rb:53]: This was changed in another PR but the comment wasn't updated

**Review by dhatch — APPROVED** [review]

File: languages/ruby/spec/oso/polar/polar_spec.rb [hunk 2]
```diff
@@ -943,4 +943,58 @@ def self.test
...
+      p = <<~POLAR
+        type f(_x: Foo{id: 1});
+        f(_x: Bar{id: 1});
+        f(_x: Baz{id: 1});
+      POLAR
+
+      subject.load_str(p)
+
+      # Should raise error
+      expect { subject.load_str('f(_x: Bad);') }.to raise_error Oso::Polar::ValidationError
```
Comment (dhatch) [inline languages/ruby/spec/oso/polar/polar_spec.rb:987]: This test is different than the python test.

**Review by dhatch — CHANGES_REQUESTED** [review]
One test is ported wrong.

**Review by dhatch — APPROVED** [review]

**Pull request merged**